testplan/runnable/base.py

```diff
@@ -1054,12 +1054,21 @@ def make_runpath_dirs(self):
 
         self._scratch = os.path.join(self._runpath, "scratch")
 
+        env_mpl_cache = os.environ.get("MPLCONFIGDIR")
+        if not env_mpl_cache:
+            self._mplcache = os.path.join(self._runpath, "matplotlib")
+            os.environ["MPLCONFIGDIR"] = self._mplcache
+
         if self.cfg.path_cleanup is False:
             makedirs(self._runpath)
             makedirs(self._scratch)
+            if not env_mpl_cache:
+                makedirs(self._mplcache)
         else:
             makeemptydirs(self._runpath)
             makeemptydirs(self._scratch)
+            if not env_mpl_cache:
+                makeemptydirs(self._mplcache)
```

Comment:
i don't think we shall always create this dir for matplotlib, 99% of time we don't need this dir. Now that the warning msg is only issued when matplotlib is actually used by pdf exporter, it's OK to leave it to user to set it.

Reply:
Removed the creation, now matplotlib automatically creates the dir. Should I also remove setting the env var to the runpath, if not set by the user?